### PR TITLE
Feature profile photo

### PR DESCRIPTION
**Pull request by Rhut-virani**
Added endpoints for uploading and deleting the profile photo
saving the new profile photo image into database
also, edit profile working correctly with data saved on the backend.

p.s. : code needs to be formatted properly, will create another pull request for that

### CODE REVIEW

**Comment by skassam21** [comment]
@Rhut-virani definitely want to gitignore env3 (you will have to remove them from being tracked and clean it) - really hard to read 1300+ changes lol

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "env3\\Scripts\\python.exe"
+}
```
Comment (skassam21) [inline .vscode/settings.json]: gitignore .vscode/* as well

File: app/api/auth.py [hunk 2]
```diff
@@ -39,12 +39,20 @@ def basic_auth_error():
...
+#     print(g.current_user)
+#     return g.current_user is not None
+
 @token_auth.verify_token
 def verify_token(token):
-    g.current_user = User.check_token(token) if token else None
-    print(g.current_user)
-    return g.current_user is not None
-
+   print("why")
```
Comment (skassam21) [inline app/api/auth.py]: remove the printing in this file (if you need to print, you should be using a logger)

File: app/api/auth.py [hunk 3]
```diff
@@ -39,12 +39,20 @@ def basic_auth_error():
...
+   g.current_user = Sitter.check_token(token) if token else None
+   if not g.current_user:
+       g.current_user = Owner.check_token(token) if token else None
```
Comment (skassam21) [inline app/api/auth.py]: I get what you are doing here, I think it's fine, but something cleaner in the future is to have one function for checking token, and to store in the token the type of object (when you get_token)

File: app/api/users.py [hunk 4]
```diff
@@ -14,17 +14,31 @@ def get_user(username):
...
+   print(request.get_json()['profile_data'])
+   profile_data = request.get_json()['profile_data']
+   birthdate = profile_data['birthdate']
+   gender = profile_data['gender']
+   about_me = profile_data['about_me']
+   location = profile_data['location']
+   charge = profile_data['rate']
+
+   token = request.get_json()['token']
+   u = collection.check_token(token)
```
Comment (skassam21) [inline app/api/users.py]: you should haven't to check token here again

File: app/api/users.py [hunk 5]
```diff
@@ -14,17 +14,31 @@ def get_user(username):
...
-def edit_profile():
-    is_sitter = request.get_json()['is_sitter']
-    collection = Sitter if is_sitter else Owner
-    token = request.get_json()['token']
-    u = collection.check_token(token)
-    return jsonify(u.to_dict(include_email=True))
+def update_profile():
+   is_sitter = request.get_json()['is_sitter']
+   collection = Sitter if is_sitter else Owner
+   print(request.get_json()['profile_data'])
```
Comment (skassam21) [inline app/api/users.py]: printing again

File: app/main/routes.py [hunk 6]
```diff
@@ -30,30 +31,34 @@ def index():
...
-            return response
+        # not using session for login or registrations 
+        # so remove current user from the code 
+        # bcoz that results in sending the same token 
+        # if current_user.is_authenticated:
+            # print('current user')
+            # print(current_user.first_name)
+            # response = jsonify(current_user.to_dict())
+            # response.status_code = 200
+            # return response
```
Comment (skassam21) [inline app/main/routes.py]: can you remove these lines of code, it's not good practice to commit a lot of commented out code

File: app/api/auth.py [hunk 7]
```diff
@@ -39,12 +39,20 @@ def basic_auth_error():
 #     return user.check_password(password)
 
 
+# @token_auth.verify_token
+# def verify_token(token):
+#     g.current_user = User.check_token(token) if token else None
+#     print(g.current_user)
+#     return g.current_user is not None
```
Comment (skassam21) [inline app/api/auth.py]: let's delete this too

File: app/main/routes.py [hunk 8]
```diff
@@ -174,20 +180,83 @@ def register():
...
+
+
+@main_bp.route('/upload_profile_image', methods=['GET', 'POST'])
+@token_auth.login_required
+def upload_file():
+    file = request.files['file']
+    file.filename = secure_filename(file.filename)
+    output   	  = upload_file_to_s3(file, current_app.config["S3_BUCKET"])
+    token = request.headers['Authorization'].split(" ")[1]
+    collection = Sitter if Sitter.check_token(token) else Owner
```
Comment (skassam21) [inline app/main/routes.py]: this should be happening in the decorator (I think it does)

File: app/main/routes.py [hunk 9]
```diff
@@ -174,20 +180,83 @@ def register():
...
+            bucket_name,
+            file.filename,
+            ExtraArgs={
+                "ACL": acl,
+                "ContentType": file.content_type
+            }
+        )
+    except Exception as e:
+        # This is a catch all exception, edit this part to fit your needs.
+        print("Something Happened: ", e)
```
Comment (skassam21) [inline app/main/routes.py]: lolll... definitely want to use a logger here

File: app/api/auth.py (hunk 2 above)
Comment (Rhut-virani) [reply app/api/auth.py]: LOL, I forgot that,
actually, as I said I need to clean up the code a lot,

prints like these are just me know which line is printing

File: app/api/auth.py (hunk 3 above)
Comment (Rhut-virani) [reply app/api/auth.py]: it seems @wangtianxing1010  and @Rhut-virani,  we both have to talk to you about the decorator and how to include the things what you are suggesting into the decorator.

File: app/api/users.py (hunk 4 above)
Comment (Rhut-virani) [reply app/api/users.py]: yes once we include the fetching of the user inside decorator we will remove this,
here it is just for the sake of making it work for the time being.

File: app/api/users.py (hunk 5 above)
Comment (Rhut-virani) [reply app/api/users.py]: please ignore all prints for the time being

File: app/main/routes.py (hunk 8 above)
Comment (Rhut-virani) [reply app/main/routes.py]: Discussing this as soon as possible

File: app/main/routes.py (hunk 9 above)
Comment (Rhut-virani) [reply app/main/routes.py]: will do it

**Comment by Rhut-virani** [comment]
Have removed env3 before this pull request from this branch but the master has it in history so as soon as this pull request is merged I think it should be removed from all branches. (huge mistake on my part)

also, pull requests are amazing and working of branches is so relieving, that you can't go back

**Pull request merged**